yarn-project/noir-contracts/contracts/reader_contract/src/main.nr

```diff
@@ -0,0 +1,80 @@
+contract Reader {
+    use dep::std::option::Option;
+    use dep::protocol_types::{
+        address::{
+            AztecAddress,
+            EthAddress,
+        },
+        abis::function_selector::FunctionSelector,
+    };
+    use dep::protocol_types;
+    use dep::aztec::log::emit_unencrypted_log;
+
+    use dep::compressed_string::{FixedsizeCompressedString, FixedsizeCompressedStringSerializationMethods};
+
+    use dep::aztec::{
+        context::Context,
+        abi,
+        abi::PrivateContextInputs,
+        hash::pedersen_hash,
+        context::PrivateContext,
+        note::{
+            note_header::NoteHeader,
+            utils as note_utils,
+        },
+        oracle::{
+            get_public_key::get_public_key as get_public_key_oracle,
+            context::get_portal_address,
+            rand::rand
+        },
+        state_vars::immutable_singleton::ImmutableSingleton,
+        log::emit_unencrypted_log_from_private,
+        types::vec::BoundedVec,
+    };
+
+    struct Storage {}
```

Comment:
do you need this?

Reply:
Thought it would actually fail without it so just left an empty one 🤷